pkg/csi/cinder/nodeserver.go

```diff
@@ -296,9 +224,7 @@ func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		}
 	}
 
-	// Try expanding the volume if it's created from a snapshot or another volume (see #1539)
-	if vol.SourceVolID != "" || vol.SnapshotID != "" {
-
+	if required, ok := volumeContext[ResizeRequired]; ok && strings.EqualFold(required, "true") {
```

Comment:
btw, it's enough to just set an empty string to the corresponding `volumeContext` key to define the resize requirement and check it like this:

```suggestion
	if , ok := volumeContext[ResizeRequired]; ok {
```

Reply:
I don't really like a free-form key definition for the [VolumeContext](https://github.com/search?q=%22req.GetVolumeContext%28%29%22&type=code). Does it make sense to use `cinder.csi.openstack.org` in a key name? CSI driver itself uses the `csi.storage.k8s.io/ephemeral` as a key to mark an ephemeral volume. For the topology we already use https://github.com/kubernetes/cloud-provider-openstack/blob/d069b1ef5d34e1400c798424f5d56b11098fb9a7/pkg/csi/cinder/driver.go#L33-L34

Reply:
Ok, I've changed it to `ResizeRequired = driverName + "/resizeRequired"`